src/Libraries/CoreNodeModels/CustomSelectionItem.cs

```diff
@@ -0,0 +1,165 @@
+using System;
+using System.ComponentModel;
+using System.Windows.Input;
+using Newtonsoft.Json;
+
+namespace CoreNodeModels
+{
+    /// <summary>
+    /// Represents an Enum item
+    /// </summary>
+    public class CustomSelectionItem
```

Comment:
why is this public? when possible we should use the *most* conservative access modifier possible to avoid exposing types and methods and having to maintain them for years.